model/simulationtests/zone_hvac_evaporative_cooler.rb

```diff
@@ -0,0 +1,125 @@
+    )
+    # An optional Second EvaporativeCooler
+    if add_secondary
+      direct_evap = OpenStudio::Model::EvaporativeCoolerDirectResearchSpecial.new(model, model.alwaysOnDiscreteSchedule)
+      zoneHVACEvaporativeCoolerUnit.setSecondEvaporativeCooler(direct_evap)
+    end
+  end
+  indirect_evap&.setName("#{z.nameString} Indirect Evaporative Cooler")
+  direct_evap&.setName("#{z.nameString} Direct Evaporative Cooler")
+  supplyAirFan.setName("#{z.nameString} Supply Fan")
+  zoneHVACEvaporativeCoolerUnit.setName("#{z.nameString} Evap Unit")
+  # zoneHVACEvaporativeCoolerUnit.resetSecondEvaporativeCooler
+  #
+  # Redoing what the default constructor does for demonstration purposes
+  # zoneHVACEvaporativeCoolerUnit.setDesignSupplyAirFlowRate(1.0)
+  zoneHVACEvaporativeCoolerUnit.autosizeDesignSupplyAirFlowRate
+  zoneHVACEvaporativeCoolerUnit.setAvailabilitySchedule(model.alwaysOnDiscreteSchedule)
+  zoneHVACEvaporativeCoolerUnit.setSupplyAirFan(zoneHVACEvaporativeCoolerUnit.supplyAirFan)
+  zoneHVACEvaporativeCoolerUnit.setFanPlacement(fan_placement)
+  zoneHVACEvaporativeCoolerUnit.setCoolerUnitControlMethod('ZoneCoolingLoadVariableSpeedFan')
+  zoneHVACEvaporativeCoolerUnit.setThrottlingRangeTemperatureDifference(1.1)
+  zoneHVACEvaporativeCoolerUnit.setCoolingLoadControlThresholdHeatTransferRate(100.0)
+  zoneHVACEvaporativeCoolerUnit.setShutOffRelativeHumidity(100.0)
+
+  # Add it to a ThermalZone
+  zoneHVACEvaporativeCoolerUnit.addToThermalZone(z)
+
+  # Rename nodes for clarity
+  z.zoneAirNode.setName("#{z.nameString} Zone Air Node")
+  # z.returnAirModelObjects.modelObjects[0].setName("#{z.nameString} Zone Return Air Node")
+  z.inletPortList.modelObjects[0].setName("#{z.nameString} Zone Air Inlet Node")
+  z.exhaustPortList.modelObjects[0].setName("#{z.nameString} Zone Air Exhaust Node")
+end
+
+# In order to produce more consistent results between different runs,
+# we sort the zones by names
+zones = model.getThermalZones.sort_by { |z| z.name.to_s }
+
+configs = [
+  { zone_name: 'DirectFirst', direct_is_first: true, add_secondary: true },
+
+  # Mimic SMStore8 from StripMallZoneEvapCoolerAutosized.idf
+  # https://github.com/NREL/EnergyPlus/blob/31e3c33467c5873371bf48b12a7318215971c315/testfiles/StripMallZoneEvapCoolerAutosized.idf#L4767-L4784
+  { zone_name: 'IndirectFirst', direct_is_first: false, add_secondary: true },
+
+  { zone_name: 'DirectOnly', direct_is_first: true, add_secondary: false },
+
+  { zone_name: 'IndirectOnly', direct_is_first: false, add_secondary: false }
+]
+fan_placements = ['BlowThrough', 'DrawThrough']
+raise 'Mismatch' unless configs.size * fan_placements.size == zones.size
+
+configs.product(fan_placements).zip(zones).each do |(config, fan_placement), z|
+  z.setName("#{config[:zone_name]} #{fan_placement} Zn")
+  make_zone_hvac_cooler(
+    z,
+    direct_is_first: config[:direct_is_first],
+    add_secondary: config[:add_secondary],
+    fan_placement: fan_placement
+  )
+end
```

Comment:
Testing 8 combinations: for each of these four, I test BlowThrough + DrawThrough

* DirectOnly
* IndirectOnly
* Direct then Indirect
* Indirect then Direct